### ISSUE TITLE
It's so perplexing: the qr/ / compiled regex slower than raw string

### ISSUE THREAD

**Issue by abdulbadii** [Perl/perl5#19405]
It's said in some Perl tutorial, manual and books that the qr/ / is a precompiled regex

but that regex slower than rawstring

```
 DB<1> use Benchmark qw(cmpthese timethis); 

$r=qr/o(?:o|k|l)\h*/ ; 
$s = 'o(?:o|k|l)\h*'  ;


cmpthese( 0, { 
 're' , sub { 'foobar'=~/f(?:$r)bar/ }, 
 str , sub { 'foobar'=~/f(?:$s)bar/ },
})

        Rate  re str
re  395785/s  -- -5%
str 415726/s  5%  --


DB<<2>> cmpthese( -7, {
 're' , sub { 'foobar'=~/f(?:$r)bar/ },
 str , sub { 'foobar'=~/f(?:$s)bar/ }, 
})

        Rate  re str
re  802743/s  -- -9%
str 886123/s 10%  --

```

**Comment by demerphq** [comment]
On Thu, 10 Feb 2022 at 01:45, abdulbadii ***@***.***> wrote:

> It's said in Perl tutorial, manual and books that the qr/ / is precompiled
> regex
>
> but thatregex slower than rawstring
>
>  DB<1> use Benchmark qw(cmpthese timethis); $r=qr/o(?:o|k|l)\h*/ ; $s = 'o(?:o|k|l)\h*'  ;cmpthese( 0, { 're' , sub { 'foobar'=~/f(?:$r)bar/ }, str , sub { 'foobar'=~/f(?:$s)bar/ }, })
>         Rate  re str
> re  395785/s  -- -5%
> str 415726/s  5%  --
>
>
Thanks for the report!

Your benchmark doesn't actually test what you think it is testing. The docs
don't say that constructing a new regex out of qr// objects will be faster
than constructing them out of strings. It says that precompiling the regex
using a qr// should allow you to compile the pattern once and reuse the
compiled object repeatedly.

It is at this point that I was expecting to be able to show you that
changing your benchmark to look like this:

$ cat /tmp/t.pl
use strict;
use warnings;
use Benchmark qw(cmpthese);

print "This is perl $]\n";

my $r=qr/o(?:o|k|l)\h*/ ;
my $s=  'o(?:o|k|l)\h*';

my $cr= qr/f(?:$r)bar/;
my $cs= qr/f(?:$s)bar/;

cmpthese( -10, {
        're'     => sub { 'foobar'=~/f(?:$r)bar/ },
        'str'    => sub { 'foobar'=~/f(?:$s)bar/ },
        're_qr'  => sub { 'foobar'=~$cr },
        'str_qr' => sub { 'foobar'=~$cs },
});
__END__

Would show that the 're_qr' and 'str_qr' were both essentially the same
performance and substantially faster than the 're' and 'str' examples.

However it does appear to me that you have found a regression:

$ for i in {1..10}; do perl /tmp/t.pl; ./perl -Ilib /tmp/t.pl; done
This is perl 5.024001
            Rate     re str_qr  re_qr    str
re     2248541/s     --    -3%    -6%   -11%
str_qr 2322514/s     3%     --    -2%    -8%
re_qr  2381942/s     6%     3%     --    -6%
str    2530367/s    13%     9%     6%     --
This is perl 5.035009
            Rate     re  re_qr str_qr    str
re     2203803/s     --    -1%    -1%    -7%
re_qr  2220606/s     1%     --    -0%    -6%
str_qr 2229074/s     1%     0%     --    -5%
str    2358398/s     7%     6%     6%     --
This is perl 5.024001
            Rate     re  re_qr str_qr    str
re     2244526/s     --    -6%    -7%   -12%
re_qr  2376990/s     6%     --    -1%    -6%
str_qr 2412935/s     8%     2%     --    -5%
str    2537322/s    13%     7%     5%     --
This is perl 5.035009
            Rate  re_qr str_qr     re    str
re_qr  2178309/s     --    -1%    -1%    -7%
str_qr 2203440/s     1%     --    -0%    -6%
re     2203803/s     1%     0%     --    -6%
str    2343045/s     8%     6%     6%     --
This is perl 5.024001
            Rate     re  re_qr str_qr    str
re     2238127/s     --    -6%    -8%   -12%
re_qr  2376617/s     6%     --    -2%    -6%
str_qr 2436713/s     9%     3%     --    -4%
str    2533974/s    13%     7%     4%     --
This is perl 5.035009
            Rate str_qr     re  re_qr    str
str_qr 2055963/s     --    -5%    -5%   -11%
re     2158262/s     5%     --    -0%    -6%
re_qr  2160275/s     5%     0%     --    -6%
str    2308136/s    12%     7%     7%     --
This is perl 5.024001
            Rate     re  re_qr str_qr    str
re     2195086/s     --    -8%   -10%   -14%
re_qr  2393593/s     9%     --    -2%    -6%
str_qr 2449850/s    12%     2%     --    -4%
str    2543776/s    16%     6%     4%     --
This is perl 5.035009
            Rate     re str_qr  re_qr    str
re     2174867/s     --    -1%    -1%    -5%
str_qr 2189577/s     1%     --    -0%    -4%
re_qr  2194985/s     1%     0%     --    -4%
str    2279144/s     5%     4%     4%     --
This is perl 5.024001
            Rate     re str_qr  re_qr    str
re     2248364/s     --    -6%    -6%   -12%
str_qr 2383343/s     6%     --    -0%    -6%
re_qr  2394529/s     7%     0%     --    -6%
str    2542187/s    13%     7%     6%     --
This is perl 5.035009
            Rate str_qr     re  re_qr    str
str_qr 2066965/s     --    -6%    -6%   -12%
re     2191052/s     6%     --    -1%    -7%
re_qr  2203419/s     7%     1%     --    -6%
str    2356156/s    14%     8%     7%     --
This is perl 5.024001
            Rate     re  re_qr str_qr    str
re     2192365/s     --    -9%   -10%   -10%
re_qr  2399187/s     9%     --    -1%    -2%
str_qr 2428911/s    11%     1%     --    -0%
str    2438035/s    11%     2%     0%     --
This is perl 5.035009
            Rate  re_qr     re str_qr    str
re_qr  2201304/s     --    -0%    -0%    -3%
re     2206480/s     0%     --    -0%    -3%
str_qr 2207641/s     0%     0%     --    -3%
str    2271499/s     3%     3%     3%     --
This is perl 5.024001
            Rate     re  re_qr str_qr    str
re     2238354/s     --    -6%    -8%   -12%
re_qr  2380859/s     6%     --    -2%    -6%
str_qr 2434037/s     9%     2%     --    -4%
str    2535838/s    13%     7%     4%     --
This is perl 5.035009
            Rate  re_qr     re str_qr    str
re_qr  2090267/s     --    -5%    -5%   -11%
re     2189487/s     5%     --    -0%    -6%
str_qr 2195086/s     5%     0%     --    -6%
str    2336030/s    12%     7%     6%     --
This is perl 5.024001
            Rate     re  re_qr str_qr    str
re     2217020/s     --    -6%    -7%   -12%
re_qr  2356199/s     6%     --    -1%    -7%
str_qr 2385398/s     8%     1%     --    -6%
str    2529122/s    14%     7%     6%     --
This is perl 5.035009
            Rate     re  re_qr str_qr    str
re     2176872/s     --    -1%    -1%    -6%
re_qr  2191051/s     1%     --    -0%    -6%
str_qr 2199174/s     1%     0%     --    -5%
str    2320899/s     7%     6%     6%     --
This is perl 5.024001
            Rate     re str_qr  re_qr    str
re     2246645/s     --    -6%    -8%   -11%
str_qr 2388089/s     6%     --    -2%    -6%
re_qr  2440926/s     9%     2%     --    -3%
str    2527647/s    13%     6%     4%     --
This is perl 5.035009
            Rate     re str_qr  re_qr    str
re     2172200/s     --    -0%    -1%    -8%
str_qr 2176242/s     0%     --    -1%    -7%
re_qr  2192886/s     1%     1%     --    -7%
str    2351685/s     8%     8%     7%     --
This is perl 5.024001
            Rate     re str_qr  re_qr    str
re     2237607/s     --    -6%    -6%   -11%
str_qr 2368497/s     6%     --    -1%    -6%
re_qr  2392371/s     7%     1%     --    -5%
str    2510369/s    12%     6%     5%     --
This is perl 5.035009
            Rate  re_qr str_qr     re    str
re_qr  2182455/s     --    -1%    -1%    -6%
str_qr 2197088/s     1%     --    -1%    -5%
re     2211919/s     1%     1%     --    -5%
str    2318708/s     6%     6%     5%     --

In every single run the 'str' case outperforms the _qr cases, and in both
perl 5.24 and 5.35. Also 5.35 is reliably slower than 5.24. Additionally
when I dig into this the regex engine is fully entering the TRIE logic even
though this case is basically a fancy way of writing a charclass.

Digging into this deeper what I can see is that perl is caching the
compiled pattern for the 'str' and 're' case after their first use.

$ cat /tmp/t.pl; ./perl -Ilib /tmp/t.pl
use strict;
use warnings;
use Benchmark qw(cmpthese);
use re 'debug';

print "This is perl $]\n";

my $r=qr/o(?:o|k|l)\h*/ ;
my $s=  'o(?:o|k|l)\h*';

my $cr= qr/f(?:$r)bar/;
my $cs= qr/f(?:$s)bar/;

cmpthese( 2, {
        're'     => sub { print "in 're'\n";     'foobar'=~/f(?:$r)bar/ },
        'str'    => sub { print "in 'str'\n";    'foobar'=~/f(?:$s)bar/ },
        're_qr'  => sub { print "in 're_qr'\n";  'foobar'=~$cr },
        'str_qr' => sub { print "in 'str_qr'\n"; 'foobar'=~$cs },
});
__END__

Compiling REx "o(?:o|k|l)\h*"
Final program:
   1: EXACT <o> (3)
   3: TRIE-EXACT[klo] (13)
      <o>
      <k>
      <l>
  13: STAR (15)
  14:   POSIXU[:blank:] (0)
  15: END (0)
anchored "o" at 0..0 (checking anchored) minlen 2
This is perl 5.035009
Compiling REx "f(?:(?^:o(?:o|k|l)\h*))bar"
Final program:
   1: EXACT <fo> (5)
   5: TRIE-EXACT[klo] (15)
      <o>
      <k>
      <l>
  15: STAR (17)
  16:   POSIXU[:blank:] (0)
  17: EXACT <bar> (19)
  19: END (0)
anchored "fo" at 0..0 floating "bar" at 3..9223372036854775807 (checking
floating) minlen 6
Compiling REx "f(?:o(?:o|k|l)\h*)bar"
Final program:
   1: EXACT <fo> (5)
   5: TRIE-EXACT[klo] (15)
      <o>
      <k>
      <l>
  15: STAR (17)
  16:   POSIXU[:blank:] (0)
  17: EXACT <bar> (19)
  19: END (0)
anchored "fo" at 0..0 floating "bar" at 3..9223372036854775807 (checking
floating) minlen 6
in 're'
Compiling REx "f(?:(?^:o(?:o|k|l)\h*))bar"
Final program:
   1: EXACT <fo> (5)
   5: TRIE-EXACT[klo] (15)
      <o>
      <k>
      <l>
  15: STAR (17)
  16:   POSIXU[:blank:] (0)
  17: EXACT <bar> (19)
  19: END (0)
anchored "fo" at 0..0 floating "bar" at 3..9223372036854775807 (checking
floating) minlen 6
Matching REx "f(?:(?^:o(?:o|k|l)\h*))bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
in 're'
Compiling REx "f(?:(?^:o(?:o|k|l)\h*))bar"
Skipping recompilation of unchanged REx "f(?:(?^:o(?:o|k|l)\h*))bar"
Matching REx "f(?:(?^:o(?:o|k|l)\h*))bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
            (warning: too few iterations for a reliable count)
in 're_qr'
Matching REx "f(?:(?^:o(?:o|k|l)\h*))bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
in 're_qr'
Matching REx "f(?:(?^:o(?:o|k|l)\h*))bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
            (warning: too few iterations for a reliable count)
in 'str'
Compiling REx "f(?:o(?:o|k|l)\h*)bar"
Final program:
   1: EXACT <fo> (5)
   5: TRIE-EXACT[klo] (15)
      <o>
      <k>
      <l>
  15: STAR (17)
  16:   POSIXU[:blank:] (0)
  17: EXACT <bar> (19)
  19: END (0)
anchored "fo" at 0..0 floating "bar" at 3..9223372036854775807 (checking
floating) minlen 6
Matching REx "f(?:o(?:o|k|l)\h*)bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
in 'str'
Compiling REx "f(?:o(?:o|k|l)\h*)bar"
Skipping recompilation of unchanged REx "f(?:o(?:o|k|l)\h*)bar"
Matching REx "f(?:o(?:o|k|l)\h*)bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
            (warning: too few iterations for a reliable count)
in 'str_qr'
Matching REx "f(?:o(?:o|k|l)\h*)bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
in 'str_qr'
Matching REx "f(?:o(?:o|k|l)\h*)bar" against "foobar"
Intuit: trying to determine minimum start position...
  doing 'check' fbm scan, [3..6] gave 3
  Found floating substr "bar" at offset 3 (rx_origin now 0)...
  doing 'other' fbm scan, [0..2] gave 0
  Found anchored substr "fo" at offset 0 (rx_origin now 0)...
  (multiline anchor test skipped)
Intuit: Successfully guessed: match at offset 0
   0 <> <foobar>             |   0| 1:EXACT <fo>(5)
   2 <fo> <obar>             |   0| 5:TRIE-EXACT[klo](15)
   2 <fo> <obar>             |   0| TRIE: State:    1 Accepted: N TRIE:
Charid:  1 CP:  6f After State:    2
   3 <foo> <bar>             |   0| TRIE: State:    2 Accepted: Y TRIE:
Charid:  0 CP:   0 After State:    0
                             |   0| TRIE: got 1 possible matches
                             |   0| TRIE matched word #1, continuing
                             |   0| TRIE: only one match left,
short-circuiting: #1 <o>
   3 <foo> <bar>             |   0| 15:STAR(17)
                             |   0| POSIXU[:blank:] can match 0 times out
of 2147483647...
   3 <foo> <bar>             |   1|  17:EXACT <bar>(19)
   6 <foobar> <>             |   1|  19:END(0)
Match successful!
            (warning: too few iterations for a reliable count)
                     Rate  re_qr     re str_qr    str
re_qr  2000000000000000/s     --     0%     0%     0%
re     2000000000000000/s     0%     --     0%     0%
str_qr 2000000000000000/s     0%     0%     --     0%
str    2000000000000000/s     0%     0%     0%     --
Freeing REx: "o(?:o|k|l)\h*"
Freeing REx: "f(?:o(?:o|k|l)\h*)bar"
Freeing REx: "f(?:(?^:o(?:o|k|l)\h*))bar"
Freeing REx: "f(?:o(?:o|k|l)\h*)bar"
Freeing REx: "f(?:(?^:o(?:o|k|l)\h*))bar"

So in some ways this is a win. Where in the past the 'str' and 're' cases
would be much slower because they would recompile the pattern each time,
they are now smart enough to cache the result and reuse it if the string
has not changed. However, that does not explain why the 're' case is
consistently slower than the 'str' case, nor why the _qr cases are not as
fast or faster than the other two.

cheers,
Yves


-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by abdulbadii** [comment]
Remind us famous switch-case C use:
it'd unexpectedly slower for 3 or so branches than that of plain if - else if
but indeed it will linearly or quadratically be faster for 40 or more ones

**Comment by demerphq** [comment]
On Fri, 11 Feb 2022 at 00:03, abdulbadii ***@***.***> wrote:

> Closed #19405 <https://github.com/Perl/perl5/issues/19405>.
>
>
>
FWIW, I am not sure this should be closed until we understand why this is
happening.

I /think/ I understand why this code path might be faster, that is it does
not have to deal with some of the edge cases that a qr// might, such as
being used in multiple contexts, but at the same time it still feels like
we can do better.

Yves



-- 
perl -Mre=debug -e "/just|another|perl|hacker/"
